podcast/2.16.md

```diff
@@ -0,0 +1,1131 @@
+
+**James Waldrop:**
+
+You can imagine that that happened, but no.
+
+**Beyang Liu:**
+
+We just got done recording this Dev Tool Time, this other stream that we're doing with [Mitchell Hashimoto](https://about.sourcegraph.com/blog/dev-tool-time-mitchell-hashimoto/?_ga=2.93485403.753704949.1643295641-1161303122.1641224283). This is really great for me to hear because he told the story of how he switched from Emacs to Vim because-
+
+**James Waldrop:**
+
+Oh, no. He left the flock.
+
+**Beyang Liu:**
+
+Anyways.
+
+**James Waldrop:**
+
+Anywho. All right, here you go. Here it's just come up and it's right here. Yeah, he should know that at Twitter, which obviously has the strongest Vi people on the planet and the strongest Emacs people on the planet, it was decided that Emacs is stronger.
+
+**Beyang Liu:**
+
+There you have it.
+
+**James Waldrop:**
+
+Yes, there you have it. It's been decided. We didn't go that far because there was no way I was wading into that debate with people. I mean, we didn't even talk about IntelliJ.
+
+So, we started at code review, okay? That's the point at which other people start interacting with your code, if you think about it that way. At least I thought about it that way. And so, we own the code review stack, we own the build system. You could think of the build system as being the earliest point. That would be a reasonable argument to make actually, now that I think about it. Build system, code review, the test infrastructure and test frameworks that people used, then the deploy tooling, how you actually got your code to production.
+
+**Beyang Liu:**
+
+Got it. And I think it was Review Board for code review and Pants for the build system?
+
+**James Waldrop:**
+
+I'm not at liberty to say so.
+
+**Beyang Liu:**
+
+Oh, sorry. Got it.
+
+**James Waldrop:**
+
+No, no, no, it's fine. Keep it in. We can talk about Pants. Twitter open sourced it. I was managing the team when we did that.
+
+So, Twitter, as many Bay Area startups do, had a fair number of refugees from Google. Maybe more than most, I would say. And so, they had all used Blaze at Google. There was no Bazel at that point in time, and so they all missed their old build tool. So, Twitter, and it turns out Facebook did this too, wrote clones of Blaze and then Google went and open sourced Bazel. And so now we have three different Blaze-inspired open source build tools that people could use if they want to.
+
+But Twitter, of course, had theirs, Pants. Has theirs, still, I believe. I don't know, I haven't checked in to see if they've moved to anything else since I left. We were on Maven when I started. Maven was causing us a lot of trouble, a lot of trouble. It couldn't really walk the dependency graph in any reasonable way, and you really need that once you're in a mono repo. That's almost job one for your build tool on a mono repo. We had these crazy ass Maven POM XMLs that were deeply nested.
```

Comment:
```suggestion
But Twitter, of course, had theirs, Pants. Has theirs, still, I believe. I don't know, I haven't checked in to see if they've moved to anything else since I left. We were on Maven when I started. Maven was causing us a lot of trouble, a lot of trouble. It couldn't really walk the dependency graph in any reasonable way, and you really need that once you're in a monorepo. That's almost job one for your build tool on a monorepo. We had these crazy ass Maven POM XMLs that were deeply nested.
```